src/utils/tests/BlueBaseModule.test.ts

```diff
@@ -56,19 +56,17 @@ describe('Utils', () => {
 			expect(obj.foo).toBe('bar');
 		});
 
-		// it('should set loaded prop of only the loaded item', async () => {
-
-		// 	// FIXME: Big bug!!!
-		// 	const module = createBlueBaseModule({ foo: 'bar' });
-		// 	const module1 = createBlueBaseModule({ foo: 'baz' });
-		// 	expect(module.loaded).toBe(false);
-		// 	expect(module1.loaded).toBe(false);
-
-		// 	const obj = await module;
+		it('should set loaded prop of only the loaded item', () => {
+			const module = createBlueBaseModule({ foo: 'bar' });
+			const module1 = createBlueBaseModule({ foo: 'baz' });
+			expect(module.loaded).toBe(false);
+			expect(module1.loaded).toBe(false);
 
-		// 	expect(obj.foo).toBe('bar');
-		// 	expect(module.loaded).toBe(true);
-		// 	expect(module1.loaded).toBe(false);
-		// });
+			module.then(obj => {
+				expect(obj.foo).toBe('bar');
+				expect(module.loaded).toBe(true);
+			});
+			expect(module1.loaded).toBe(false);
```

Comment:
What happens if you move this test inside the `then` callback? @omarnaeem0

Reply:
@artalat sir are you refering to the line 69?

Reply:
yes